### ISSUE TITLE
(aws-appsync): RdsDataSource misses option to set dataBaseName and expects incorrect cluster

### ISSUE THREAD

**Issue by Simon-TechForm** [aws/aws-cdk#12503]
It seems like there's no way to add the databasename to a RdsDataSource, which results in a "No database selected" error when running a resolver.

Besides that, RdsDataSource expects a DatabaseCluster and not a ServerlessCluster.

https://docs.aws.amazon.com/cdk/api/latest/docs/aws-appsync-readme.html#aurora-serverless seems outdated as well, and should probably be updated when a solution for the above-mentioned problems have been found. I will open a seperate issue for the doc issue, if/when this becomes relevant.

### Reproduction Steps

```
const api = new GraphqlApi(stack, 'api', {
  name: 'graphqlApi',
  schema: Schema.fromAsset('./testSchema.graphql'),
});

const serverless = new ServerlessCluster(stack, 'cluster', {
  engine: DatabaseClusterEngine.AURORA_MYSQL,
  vpc: new Vpc(stack, 'vpc'),
});

api.addRdsDataSource('rdsDS', serverless, serverless.secret!);
```

### What did you expect to happen?

addRdsDataSource to include a prop for databaseName and to expect IServerlessCluster.

### What actually happened?

There's no prop for databaseName and it expects IDatabaseCluster.

### Environment

  - **CDK CLI Version : 1.83.0**
  - **Framework Version: 1.83.0**
  - **Node.js Version: 14.5.4**
  - **OS: MacOs 10.15.7**
  - **Language (Version): TypeScript (3.9.7)**
--- 

This is :bug: Bug Report

**Comment by Simon-TechForm** [comment]
Having had a look at the relevant code, I might be able to take a stab at it, if someone could confirm that I'm not missing anything obvious.

**Comment by BerndWessels** [comment]
this is still not resolved in `"@aws-cdk/aws-appsync": "^1.89.0",` and renders `RdsDataSource` useless.
